src/slack_model.jl

```diff
@@ -231,3 +278,105 @@ function hprod!(nlp :: SlackModel, x :: AbstractVector, v :: AbstractVector,
   @views hv[n+1:nlp.meta.nvar] .= 0
   return hv
 end
+
+function residual(nlp :: SlackNLSModel, x :: AbstractVector)
+  return residual(nlp.model, x[1:nlp.model.meta.nvar])
+end
+
+function residual!(nlp :: SlackNLSModel, x :: AbstractVector, Fx :: AbstractVector)
+  return residual!(nlp.model, x[1:nlp.model.meta.nvar], Fx)
+end
+
+function jac_residual(nlp :: SlackNLSModel, x :: AbstractVector)
+  n = nlp.model.meta.nvar
+  ns = nlp.meta.nvar - n
+  ne = nlp.nls_meta.nequ
+  Jx = jac_residual(nlp.model, x[1:n])
+  if issparse(Jx)
+    return [Jx spzeros(ne, ns)]
+  else
+    return [Jx zeros(ne, ns)]
+  end
+end
+
+function jprod_residual(nlp :: SlackNLSModel, x :: AbstractVector, v :: AbstractVector)
+  return jprod_residual(nlp.model, x[1:nlp.model.meta.nvar],
+                        v[1:nlp.model.meta.nvar])
+end
+
+function jprod_residual!(nlp :: SlackNLSModel, x :: AbstractVector, v :: AbstractVector, Jv :: AbstractVector)
+  return jprod_residual!(nlp.model, x[1:nlp.model.meta.nvar],
+                         v[1:nlp.model.meta.nvar], Jv)
+end
+
+function jtprod_residual(nlp :: SlackNLSModel, x :: AbstractVector, v :: AbstractVector)
+  n = nlp.model.meta.nvar
+  ns = nlp.meta.nvar - n
+  return [jtprod_residual(nlp.model, x[1:n], v); zeros(ns)]
+end
+
+function jtprod_residual!(nlp :: SlackNLSModel, x :: AbstractVector, v :: AbstractVector, Jtv :: AbstractVector)
+  n = nlp.model.meta.nvar
+  ns = nlp.meta.nvar - n
+  @views jtprod_residual!(nlp.model, x[1:n], v, Jtv[1:n])
```

Comment:
Sometimes you use `@views` and sometimes not. I'm not entirely sure when it is necessary.

Reply:
`@views` is the same as adding `@view` individually. `@view` doesn't work on assignments, so when there is only one I use `@view`. May have missed some places, let me check again.